src/app.cpp

```diff
@@ -81,11 +81,35 @@ class simulationManager {
         }
     }
 
+    // add the step function with the input variables
+    void stepSimulation(py::dict input_dict) {
+        for (auto item : input_dict) {
+            std::string key = py::str(item.first);
+            Eigen::MatrixXd values = item.second.cast<Eigen::MatrixXd>();
+
+            if (values.cols() == 1){
+                values.transposeInPlace();
+            }
+            if (key == "position"){
+                soft_robots->applyPositionBC(values);
+            } else if (key == "twist") {
+                soft_robots->applyTwistBC(values);
+            } else if (key == "curvature") {
+                soft_robots->applyCurvatureBC(values);
+            }
```

Comment:
Use a switch statement with default handling for incorrect keys.

Reply:
we can not use switch for string variable